### ISSUE TITLE
Add a field to scanner_info so that we can specify minimum number of distinct bytes necessary in an sbuf to run a scanner

### ISSUE THREAD

**Issue by simsong** [simsong/bulk_extractor#296]
We don't need to run scan_acct unless there are 4 distinct bytes in the sbuf.
We don't need to run scan_aes unless there are .. how many bytes in the sbuf? We need to find out how many distinct bytes a scheduled key typically has
- [ ] Update scan_aes so that we can find out how many distinct bytes are associated with a random aes_key and with all zeros but one and the key 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15
- [ ] Update scan_info so that we can specify to not run a scanner if the amount of distinct bytes isn't present.
- [ ] https://github.com/simsong/be13_api/issues/59 so that we count the number of distinct bytes once and only once.

**Comment by simsong-codex** [comment]
## Current status

The core request is **already implemented**:

- `scanner_info::min_distinct_chars` exists and defaults to **4**.
- `scanner_set::process_sbuf(scanner, ...)` bypasses a scanner when the sbuf has fewer distinct bytes.
- `sbuf_t::get_histogram()` is lazy and cached, so the 256-bin histogram (and therefore the distinct-byte count) is built once per sbuf, not once per scanner.

Therefore this issue is not needed for a new API or a shared-count implementation. What remains useful is a small, evidence-driven policy adjustment for scanners whose semantics make low-diversity input meaningful.

## Recommendation by built-in scanner

These values apply to the whole sbuf, before a scanner is called. A value of 4 is the existing default and should be retained unless a checked-in regression fixture proves a valid low-diversity input.

| Scanner | Recommended minimum | Assessment |
|---|---:|---|
| `accts` | 4 | Retain default; account patterns need varied text. |
| `aes` | 4 | Retain default; valid expanded AES schedules are not constant. Do not raise without the proposed schedule corpus. |
| `base16` | 4 | Retain default; lower values create broad repetitive-text work with little forensic value. |
| `base64` | 4 | Retain default for the same reason as base16. |
| `email` | 4 | Retain default; an address requires varied syntax. |
| `gps` | 4 | Retain default; supported coordinate forms are varied. |
| `vin` | 4 | Retain default; a VIN has varied alphanumeric content. |
| `elf` | 4 | Retain default; ELF identification is already four distinct bytes. |
| `exif` | 4 | Retain default; successful JPEG/Exif parsing requires more than a sparse marker. |
| `exiv2` | 4 | Retain default; metadata extraction requires a structured image. |
| `evtx` | 4 | Retain default; EVTX structures are nontrivial binary records. |
| `facebook` | 4 | Retain default; the recognized artifacts are structured text. |
| `gzip` | 4 | Retain default; a usable gzip member has a nontrivial header/stream. |
| `hiberfile` | 4 | Retain default; hibernation structures are nontrivial. |
| `httplogs` | 4 | Retain default; a request/log line is varied text. |
| `json` | 4 | Retain default; this scanner emits only completed objects/arrays with at least two commas. |
| `kml` | 4 | Retain default; KML/XML markup is varied text. |
| `msxml` | 4 | Retain default; XML parsing needs varied markup. |
| `net` | 4 | Retain default; packet validation rejects degenerate addresses/headers. |
| `ntfsindx` | 4 | Retain default; `INDX` records are structured binary data. |
| `ntfslogfile` | 4 | Retain default; log records are structured binary data. |
| `ntfsmft` | 4 | Retain default; `FILE` records are structured binary data. |
| `ntfsusn` | 4 | Retain default; USN records are structured binary data. |
| `outlook` | 4 | Retain default; Outlook structures are nontrivial binary data. |
| `pdf` | 4 | Retain default; a usable PDF header/body is varied text. |
| `rar` | 4 | Retain default; RAR signatures and member metadata are nontrivial. |
| `rtti` | 4 | Retain default; RTTI artifacts require structured/string data. |
| `sqlite` | 4 | Retain default; the SQLite header is nontrivial. |
| `utmp` | 4 | Retain default; valid records contain multiple fields and values. |
| `vcard_carved` | 4 | Retain default; vCard syntax is varied text. |
| `windirs` | 4 | Retain default; FAT/NTFS directory structures are nontrivial. |
| `winlnk` | 4 | Retain default; link headers/GUIDs are nontrivial. |
| `winpe` | 4 | Retain default; a usable PE has varied DOS/PE headers. |
| `winprefetch` | 4 | Retain default; Prefetch records are nontrivial. |
| `xor` | 4 | Retain default; XOR-recursing a constant sbuf only creates another constant sbuf. |
| `zip` | 4 | Retain default; ZIP headers and entries are nontrivial. |
| `find` | **0** | Change. User-supplied patterns may intentionally match constant or near-constant data, including byte patterns. |
| `wordlist` | **0** | Change. Configured word/string matching should not be suppressed merely because the input is repetitive. |

Optional, currently unregistered Lightgrep scanners should retain 4 for `accts_lg`, `base16_lg`, `email_lg`, and `gps_lg`; use 0 for the generic `scan_lightgrep` entry point if it is enabled for arbitrary user patterns. Test-only `sha1_test` should use 0 because it promises to hash every sbuf.

## Complexity and next step

Changing a value is a one-line scanner initialization change, but each exception needs a regression test that proves the low-diversity input is meaningful. The recommended follow-up is small: set `find` and `wordlist` to 0, add tests for a repeated-byte/user-pattern case and a repeated-character word/string case, and leave all other production scanners at 4. The AES characterization task is not required to preserve the safe current value, but is required before considering a higher threshold.

Recommendation: keep this issue open only for that focused `find`/`wordlist` change; create a separate benchmark/fixture issue before proposing any wider threshold tuning.
